File: spoj_ashu1461/accepted/javac.py
def function(a):
	y=''
	i=0
	while i < len(a):
		if a[i]!='_':

			y=y+a[i]
			if ord(a[i-1])>122 or ord(a[i-1]) <97:
				print 'Error!'
				exit()
		else:
			y=y+chr(ord(a[i+1])-32)
			i=i+1
		i=i+1
	print y		
def function2(a):
	y=''
	for i in range(0,len(a)):
		if ord(a[i])<95:
				y=y+'_'+chr(ord(a[i])+32)
		else:
				y=y+a[i]
	print y
a=str(raw_input())
if a[0]== '_' or a[len(a)-1]== '_' or ord(a[0]) < 95:
	print 'Error!'
	exit()
flag=0
flag1=0
for i in range(0,len(a)):
	if a[i]=='_' and flag==0:
		flag=1
	elif ord(a[i]) < 95 and flag1==0:
		flag1=1
	if flag==1 and flag1==1:
		print 'Error!'
		exit()
if flag==1:
	function(a)
else:
	function2(a)

File: spoj_ashu1461/accepted/javac.c
#include<stdio.h>
#include<string.h>
int main()
{
	char s[1000];
	while(scanf("%s",s)!=EOF)
	{
	int sta=strlen(s);
	int A[sta],i;
	int flag=0,flag1=0;
	for(i=0;i<sta;i++)
	{
		if(s[i]=='_')
		{
			if(i==0||i==sta-1)
				flag1=1;
			else 
			{
				if(s[i-1]=='_')
					flag1=1;
			}
			// check whether the adjacent elements are related or not ! 

			flag=1;
		}
		else if(65<=s[i]&&s[i]<=90)
		{
			if(i==0)
				flag=1;
//			else 
//			{
//				if(65<=s[i-1]&&s[i-1]<=90)
//					flag=1;
//			}
			flag1=1;
		}
	}
	if(flag1==1&&flag==1)
	{
		printf("Error!\n");
	}
	else
	{
	for(i=0;i<sta;i++)
	{
		if(65<=s[i]&&s[i]<=90)
		{
//			printf("here %c\n",s[i],s[i]);
			printf("_%c",s[i]+32);
	//		i++;
		}
		else if(s[i]=='_')
		{
			printf("%c",s[i+1]-32,s[i+1]-32);
			i++;
		}
		else
		{
			printf("%c",s[i]);
		}

	}
	printf("\n");
	}
	}
	return 0;
}
